### ISSUE TITLE
Working on local network but when running on remote I get this error 'EMAXBUFFER: An Upstream (`file`) for anything but the tinies files

### ISSUE THREAD

**Issue by FranciZ** [balderdashy/skipper#108]
It all works perfectly fine when I upload the file with the server running locally but as soon as I put it on a remote server I get the following error 'EMAXBUFFER: An Upstream (`file`) . The only files that successfully upload are 350 bytes or less. I've tried it both with proxy(nginx) and no proxy (IP:PORT) in front of the app with no success.

I'm using Express 4 for the server.

After the EMAXBUFFER error the request in the browser continues but the file is not being streamed anywhere.

The file is there, if I for example put 'avatar' as the name I get a different error (timeout). Also if I inspect the request made by the browser the file is there and the same setup works if I have the server running locally it's just when I put it on a remote server that the EMAXBUFFER issue pops up.

```javascript
return new Promise(function(resolve, reject){

        req.file('file')
            .upload(function (err, uploadedFiles) {
                if (err) {
                    APP.logger.error(err);
                    reject();
                }
                else {
                    resolve();
                }
            });

    });
```

**Comment by sailsbot** [comment]
Thanks for posting, @FranciZ.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by harshmaur** [comment]
I am experiencing the same problem

**Comment by KarovInal** [comment]
Hi @FranciZ , @harshmaur
I had a similar problem with promises.
And I fixed it with Promise.all:
`
      const uploadPromise = fieldName => new Promise((resolve, reject) => {
        req.file(fieldName).upload({
          dirname: path.resolve(sails.config.appPath, `.tmp/competitions/${newCompetitionID}`)
        }, (err, file) => {
          if (err) {
            return reject(err);
          }

          return resolve(file);
        });
      });

      await Promise.all([
        uploadPromise('coverCompetition'),
        uploadPromise('documentCompetition')
      ]);
`
